relay-sampling/src/lib.rs

```diff
@@ -427,57 +399,95 @@ impl SamplingRule {
     ///
     /// Non-decaying rules are always active. Decaying rules are active if they are
     /// neither idle nor expired.
-    fn is_active(&self, now: DateTime<Utc>) -> bool {
-        self.time_range.contains(now)
-    }
-
-    /// Returns the sample rate of this specific sampling rule based on the existence of the
-    /// decaying function.
-    ///
-    /// In case of any problems that render the decaying function impossible to compute, we are
-    /// going to fallback to the base sample rate.
-    pub fn get_sample_rate(&self, now: DateTime<Utc>) -> f64 {
-        if let Some(params) = self.get_decaying_function_params(now) {
-            self.decaying_fn.call(params)
-        } else {
-            self.sample_rate
+    fn is_active(&self, now: DateTime<Utc>) -> Option<MatchingRule> {
+        match self.decaying_fn {
+            DecayingFunction::Linear {
+                decayed_sample_rate,
+            } => {
+                if let TimeRange {
+                    start: Some(start),
+                    end: Some(end),
+                } = self.time_range
+                {
+                    if self.sample_rate > decayed_sample_rate && start < end && now >= start {
```

Comment:
Because `time_range.contains` is not called in this branch anymore, we need to also check that `now` is `< end` (feel free to change `<=` to `<` and vice versa). 
```suggestion
                    if self.sample_rate > decayed_sample_rate && start <= now && now < end {
```

We do not need to check `start < end` explicitly then, because it follows from `start <= now < end`.

Reply:
Apparently this was not caught by any test case. Maybe we should add one?

Reply:
Actually this was done with a purpose but for the previous implementation, now we should explicitly verify like you said because the check is in `is_active`.